src/Core/src/Platform/iOS/KeyboardAutoManager.cs

```diff
@@ -0,0 +1,162 @@
+﻿/*
+ * This class is adapted from IQKeyboardManager which is an open-source
+ * library implemented for iOS to handle Keyboard interactions with
+ * UITextFields/UITextViews. Link to their MIT License can be found here:
+ * https://github.com/hackiftekhar/IQKeyboardManager/blob/7399efb730eea084571b45a1a9b36a3a3c54c44f/LICENSE.md
+ */
+
+using System.Collections.Generic;
+using UIKit;
+
+namespace Microsoft.Maui.Platform;
+
+internal static class KeyboardAutoManager
+{
+	// you can provide a topView argument or it will try to find the ContainerViewController
+	internal static void GoToNextResponderOrResign(UIView view, UIView? topView = null)
+	{
+		if (!view.CheckIfEligible())
+		{
+			view.ResignFirstResponder();
+			return;
+		}
+
+		var targetSuperview = topView ?? view.GetViewController()?.GetContainerViewController();
```

Comment:
These two calls feel slightly redundant. 

If we make `GetViewController` take a generic type for what type of `VC` is being searched for would that let us only have the one method? 

I realize `GetContainerViewController` is returning the actual `View` but I think it's fine to just pull that off the `VC` here

Reply:
The difference I see is that one uses superviews and the other uses NextResponder (which I guess are pretty similar), but putting them together seems to work!